packages/next-auth/src/providers/oauth.ts

```diff
@@ -71,12 +57,12 @@ export type TokenEndpointHandler = EndpointHandler<
      * Parameters extracted from the request to the `/api/auth/callback/:providerId` endpoint.
      * Contains params like `state`.
      */
-    params: CallbackParamsType
+    params: URLSearchParams
     /**
      * When using this custom flow, make sure to do all the necessary security checks.
-     * Thist object contains parameters you have to match against the request to make sure it is valid.
+     * This object contains parameters you have to match against the request to make sure it is valid.
      */
-    checks: OAuthChecks
```

Comment:
I wonder if this is still necessary. I couldn't find the equivalent in `oauth4webapi`

Reply:
It's basically the way to tell if the provider wants to use `state`, `pkce`, both or `none`. (`nonce` support is WIP #4100, but I haven't had the time to have a look yet).

If omitted, our default is to use `state` currently.